great_expectations/experimental/datasources/sqlite_datasource.py

```diff
@@ -16,62 +16,74 @@
 
 from great_expectations.experimental.datasources.sql_datasource import (
     BatchSortersDefinition,
-    ColumnSplitter,
     DatetimeRange,
     QueryAsset,
+    SQLAsset,
     SQLDatasource,
     SQLDatasourceError,
+    SqlYearMonthSplitter,
     TableAsset,
     _query_for_year_and_month,
-    _SQLAsset,
 )
 
 
-class _SplitterMixin:
+class SqliteTableAsset(TableAsset):
```

Comment:
Hopefully I'm not missing anything regarding the benefits of these mixins, but it doesn't seem scalable as we will likely continue to require dialect-specific `SQLAsset`s.

Reply:
Also, moving to inheritance gets rid of the `isinstance` checks and `type: ignore`s.

Reply:
It was about removing verbosity but this looks better to me.